pyhf/pdf.py

```diff
@@ -30,6 +31,7 @@ def from_spec(cls,spec,poiname = 'mu', qualify_names = False):
                     modifier = instance.add_or_get_modifier(channel, sample, modifier_def)
                     modifier.add_sample(channel, sample, modifier_def)
                     modifiers.append(modifier_def['name'])
+                    sample['modifiers_by_type'].setdefault(modifier_def['type'],[]).append(modifier_def['name'])
```

Comment:
How is this affected by `qualify_names`?

Reply:
once we decide on a new name there is no difference viz. a spec that already had unique names (not we update the modifer_def structure `modifier_def['name'] = fullname`)

Reply:
`qualify_names` was needed because we had conflicts between modifiers of different types having the same names. I guess when we fix this, then `qualify_names` is actually not needed anymore, since we were trying to fix what we thought was a bug, but is apparently a feature?

Reply:
that's right

Reply:
#275 will handle the deprecation.